extensions/amp-analytics/0.1/linker.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {Services} from '../../../src/services';
+import {addParamToUrl} from '../../../src/url';
+import {crc32String} from './crc32';
+
+/** @const {string} */
+const DELIMITER = '~';
+
+/** @const {number} */
+const TIME_TOLERANCE = 60000; // 60 seconds.
+
+/** @typedef {{
+    url: string,
+    key: string,
+    version: string,
+    pairs: (Object<string, string>|undefined)
+  }} */
+export let LinkerConfigDef;
+
+
+/**
+ * Creates a new query string parameter to be used for cross-domain ID syncing.
+ * &<paramName>=<version>~<checksum>~<key1>~<value1>~<key2>~<value2>...
+ */
+export class Linker {
```

Comment:
Was this pun intended? =P 
https://en.wikipedia.org/wiki/Linker_(computing)
It's possible that this could be slightly confusing to people from an OS background...

Reply:
Nope :). This is already what GA had named this functionality.

Reply:
GA =.="